### PR TITLE
add support for app:// in localStorage

### PR DESCRIPTION
**Pull request by jeff-zucker**

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
Can you fix the build?

**Comment by jeff-zucker** [comment]
Sorry, working on it.

**Comment by jeff-zucker** [comment]
I'm afraid I am not getting very far.  I can build and then use the bundle in an HTML page but I can never get the tests to  succeed.  I have the same problem with tests when I clone, install, and test your master branch.  With either your master or my fork, I get: 
```
 PASS  src/__test__/url-util.spec.js
 PASS  src/__test__/popup.spec.js
 FAIL  src/__test__/solid-auth-client.spec.js
  ● Console
    console.warn src/webid-oidc.js:22
      Error logging in with WebID-OIDC
 ```

**Comment by jeff-zucker** [comment]
@RubenVerborgh I think I have fixed the PR but I have a more basic question.  If we require solid-rest-browser in solid-auth-client, the consumer will need to have solid-rest-browser locally installed where the solid-auth-client can find it.  Users will not be able to use CDN for either package.  (or do I have this wrong?) If I'm correct in this, wouldn't it be better to just let the consumer pull in a window.appfetch function with a CDN for solid-rest-browser which could then be used in solid-auth-client without a require?

File: src/solid-auth-client.js [hunk 1]
```diff
@@ -23,6 +23,9 @@ export default class SolidAuthClient extends EventEmitter {
   _pendingSession: ?Promise<?Session>
 
   fetch(input: RequestInfo, options?: RequestOptions): Promise<Response> {
+    if (input.toString().startsWith('app:')) {
+      return require('solid-rest-browser/src').appfetch(input, options)
```
Comment (RubenVerborgh) [inline src/solid-auth-client.js]: The `/src` strikes me as weird

**Comment by RubenVerborgh** [comment]
>  If we require solid-rest-browser in solid-auth-client, the consumer will need to have solid-rest-browser locally installed where the solid-auth-client can find it. Users will not be able to use CDN for either package.

Not locally: they can use anything that exposes `window['solid-rest-browser`]` as it is currently coded.

> If I'm correct in this, wouldn't it be better to just let the consumer pull in a window.appfetch function with a CDN for solid-rest-browser which could then be used in solid-auth-client without a require?

Could be; then you want to change https://github.com/solid/solid-auth-client/pull/121/files#diff-bfe8e6023b4bd1049ee37f8e6ff32a7bR33 into `'solid-rest-browser': "window"`.

**Comment by jeff-zucker** [comment]
Thanks much for the responses, I'll dig back into it.

**Comment by jeff-zucker** [comment]
:-( try again ...

**Comment by jeff-zucker** [comment]
I have recently got solid-rest working with [browserFS](https://github.com/jvilk/BrowserFS) which supports native file system API, indexDB, Dropbox, and many other backends.  I'm submitting a PR to hand off app:// URIs.  It expects the user to bring their own solid-rest so no import or dependencies are needed.

Closing this and reopening with new PR.

**Pull request closed without merging**